translate/pytest_translate

```diff
@@ -2,63 +2,37 @@
 #
 # Translate a python test to another language.
 #
-import enum
```

Comment:
Why did you remove the language differentiation logic?  Isn't the intention to support multiple languages?  `pytest_translate` is supposed to be able to translate multiple languages, not just `C#`.

Reply:
Yes, but no other implementations exist yet.

Reply:
Right but the logic is there to detect the language from the file extension. If you try to translate to another language it will detect it and give you a nice error message. Why remove that functionality?

Reply:
With the current implementation, if I run
```
./translate/pytest_translate tests/test_simple.py foo.java
```
I'll get this error message:
```
ERROR: unknown file extension from 'foo.java'
```

With your removal, we are no longer detecting the language, we will just generate `C#` code to put in `foo.java`.  If you don't want to use the file extension to detect the language, then we should add a command-line option, we shouldn't remove the language option because now the user will have unexpected results, like generate `C#` in a Java source file.

Reply:
It's an empirical question as to whether YAGNI and in a sense also an optimisation problem in the face of uncertainty and through time where the right choice depends on the consequence of removing code that actually you are going to need versus cluttering up the code with maybes.

The dimension of those choices within a commercial environment is dollars though the expression may be a deeply technical one.  Ie quality and clutter - they are important and do have often neglected dollar consequences.

But so does deleting stuff you definitely are going to need.  The cost of the latter depends on the cost of putting the stuff that was deleted back when it turns out to be needed, just as you reckoned.

Much depends on the context too.  If requirements are specified by managers who in truth do not really know what's going to be needed, then YAGNI is a sensible default.  So it's probably a sensible default in almost every context, just not absolutely every context and there are exceptions.

Here, on the whole, I think I know better than users and practitioners on the business side what they are going to need, with an unfair advantage coming from the fact that I have quite some say over what they are going to be needing!

So it's my opinion that a different balance would be the right one in this context.  I haven't been able to really succeed in having a discussion about that though, one reason being that remote does make it harder to recognise the context and another being that negative slack makes it harder to do the necessary work in communicating the context.

Reply:
I won't argue that we are 100% going to support multiple languages.  However, I do think that we should detect this use case and error rather than blindly generate `C#` code in a non `.cs` file:
```
./pytest_translate test_simple.py foo.java
```

The code being removed here is only detecting the language and printing an error when an unsupported language is detected, it does not contain implementation for any other language.

Reply:
It's still the only implementation that exists. At most, it should raise an exception if the file extension isn't `.cs`, and even if different backends existed, the enum wasn't necessary.